packages/dd-trace/src/config.js

```diff
@@ -181,19 +181,17 @@ ken|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)
 |[\\-]{5}BEGIN[a-z\\s]+PRIVATE\\sKEY[\\-]{5}[^\\-]+[\\-]{5}END[a-z\\s]+PRIVATE\\sKEY|ssh-rsa\\s*[a-z0-9\\/\\.+]{100,}`
     )
 
-    const sampler = (options.experimental && options.experimental.sampler) || {}
     const ingestion = options.ingestion || {}
     const dogstatsd = coalesce(options.dogstatsd, {})
-
-    Object.assign(sampler, {
+    const sampler = {
       sampleRate: coalesce(
         options.sampleRate,
         ingestion.sampleRate,
-        sampler.sampleRate,
         process.env.DD_TRACE_SAMPLE_RATE
       ),
-      rateLimit: coalesce(ingestion.rateLimit, sampler.rateLimit, process.env.DD_TRACE_RATE_LIMIT)
-    })
+      rateLimit: coalesce(options.rateLimit, ingestion.rateLimit, process.env.DD_TRACE_RATE_LIMIT),
+      samplingRules: coalesce(options.samplingRules, process.env.DD_TRACE_SAMPLING_RULES)
```

Comment:
I can't find where the environment variable is parsed. That also doesn't seem to be tested.

Reply:
Ah yeah that's my bad, didn't think about this env var being set as a string that would represent a JSON array, I'll add this in w tests 👍